lib/sourceFlags.js

```diff
@@ -0,0 +1,111 @@
+        flags.productionMode = true
+        break
+      case '-cores':
+        logger.warn(`Detected use of "${arg}" command line argument. This will soon be deprecated in favor of "--cores" / "-c"`.yellow.bold)
+        flags.cores = array[index + 1]
+        break
+      case 'enable-validator':
+        logger.warn(`Detected use of "${arg}" command line argument. This will soon be deprecated in favor of "--enable-validator" / "--html-validator" / "-h"`.yellow.bold)
+        flags.enableValidator = true
+        break
+      case 'disable-validator':
+        logger.warn(`Detected use of "${arg}" command line argument. This will soon be deprecated in favor of "--disable-validator" / "--raw" / "-r"`.yellow.bold)
+        flags.disableValidator = true
+        break
+      case 'detach-validator':
+        logger.warn(`Detected use of "${arg}" command line argument. This will soon be deprecated in favor of "--background-validator" / -b"`.yellow.bold)
+        flags.backgroundValidator = true
+        break
+      case 'attach-validator':
+        logger.warn(`Detected use of "${arg}" command line argument. This will soon be deprecated in favor of "--attach-validator" / "-a"`.yellow.bold)
+        flags.attachValidator = true
+        break
+
+      case '--development-mode':
+      case '--dev':
+      case '-d':
+        flags.developmentMode = true
+        break
+      case '--production-mode':
+      case '--prod':
+      case '-p':
+        flags.productionMode = true
+        break
+      case '--cores':
+      case '-c':
+        flags.cores = array[index + 1]
+        break
+      case '--enable-validator':
+      case '--html-validator':
+      case '-h':
+        flags.enableValidator = true
+        break
+      case '--disable-validator':
+      case '--raw':
+      case '-r':
+        flags.disableValidator = true
+        break
+      case '--background-validator':
+      case '-b':
+        flags.backgroundValidator = true
+        break
+      case '--attach-validator':
+      case '-a':
+        flags.attachValidator = true
+        break
+      default:
+        if (arg.startsWith('-') && !arg.startsWith('--') && !blacklist.includes(arg)) {
+          arg.split('').forEach((smallArg) => {
+            switch (smallArg) {
+              case 'd':
+                flags.developmentMode = true
+                break
+              case 'p':
+                flags.productionMode = true
+                break
+              case 'c':
+                flags.cores = array[index + 1]
+                break
+              case 'h':
+                flags.enableValidator = true
+                break
+              case 'r':
+                flags.disableValidator = true
+                break
+              case 'b':
+                flags.backgroundValidator = true
+                break
+              case 'a':
```

Comment:
we need logic for what happens if these get chained in incoherent ways. what if i pass `-dphrba`?